src/DataBus/AzureDataBusPersistence.cs

```diff
@@ -14,11 +20,63 @@ protected override void Setup(FeatureConfigurationContext context)
+            ThrowIfConnectionStringAndTokenProviderSpecified(dataBusSettings);
 
-            var dataBus = new BlobStorageDataBus(cloudBlobClient.GetContainerReference(dataBusSettings.Container), dataBusSettings, new AsyncTimer());
+            var container = CreateCloudBlobContainer(dataBusSettings);
+
+            var dataBus = new BlobStorageDataBus(container, dataBusSettings, new AsyncTimer());
 
             context.Container.ConfigureComponent(b => dataBus, DependencyLifecycle.SingleInstance);
         }
+
+        static void ThrowIfConnectionStringAndTokenProviderSpecified(DataBusSettings dataBusSettings)
+        {
+            if (dataBusSettings.RenewalTimeBeforeTokenExpires != null && dataBusSettings.UserProvidedConnectionString)
+            {
+                throw new Exception("More than one authentication method to Azure Service was supplied (using connection string and Managed Identity). Use one method only.");
+            }
+        }
+
+        static CloudBlobContainer CreateCloudBlobContainer(DataBusSettings dataBusSettings)
+        {
+            CloudBlobContainer container;
+
+            // Attempt managed identity identity first
+            if (!string.IsNullOrWhiteSpace(dataBusSettings.StorageAccountName))
+            {
+                var azureServiceTokenProvider = new AzureServiceTokenProvider();
+                var state = (azureServiceTokenProvider, dataBusSettings.RenewalTimeBeforeTokenExpires);
+                var tokenAndFrequency = TokenRenewerAsync(state, CancellationToken.None).GetAwaiter().GetResult();
+                var tokenCredential = new TokenCredential(tokenAndFrequency.Token, TokenRenewerAsync, azureServiceTokenProvider, tokenAndFrequency.Frequency.Value);
+                var storageCredentials = new StorageCredentials(tokenCredential);
+                var containerPath = $"https://{dataBusSettings.StorageAccountName}.blob.core.windows.net/{dataBusSettings.Container}";
+
+                container = new CloudBlobContainer(new StorageUri(new Uri(containerPath)), storageCredentials);
+            }
+            else // fallback to connection string
+            {
+                var cloudBlobClient = CloudStorageAccount.Parse(dataBusSettings.ConnectionString).CreateCloudBlobClient();
+                container = cloudBlobClient.GetContainerReference(dataBusSettings.Container);
+            }
+
+            return container;
+        }
+
+        static async Task<NewTokenAndFrequency> TokenRenewerAsync(object state, CancellationToken token = default)
+        {
+            var (azureServiceTokenProvider, renewal) = (ValueTuple<AzureServiceTokenProvider, TimeSpan>)state;
+
+            // Use the same token provider to request a new token.
+            var result = await azureServiceTokenProvider.GetAuthenticationResultAsync("https://storage.azure.com", cancellationToken: token).ConfigureAwait(false);
```

Comment:
> https://storage.azure.com

Is this value fixed for all datacenters? e.g. same for  a public DC and a cloud gov DC?

Reply:
Discussed and concluded to pass an additional (optional) parameter to specify the storage domain name. which could help us with cloud DCs that are not public (US government, China, Germany, etc). The default will be set to `windows.net` for the public cloud, `usgovcloudapi.net` for the US government cloud, etc.  This will require to change the URI used by [`AzureServiceTokenProvider`](https://github.com/Particular/NServiceBus.DataBus.AzureBlobStorage/pull/91/files#diff-92584a4098652b2fd300a137278c018fR70) to the root URI of the storage account. E.g. `https://<account-name>.blob.core.<domain-name>`

Reply:
Raised an [advisory](https://www.yammer.com/azureadvisors/#/Threads/show?threadId=573854388862976) question to hear from Storage team. Would be good to have a reference to the document where those domains are